kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/BigQueryWriter.java

```diff
@@ -124,10 +126,13 @@ public void writeRows(PartitionedTableId table,
               rows.size() - failedRowsMap.size(), failedRowsMap.size());
           // update insert rows and retry in case of partial failure
           rows = getFailedRows(rows, failedRowsMap.keySet(), table);
+          // trace SinkRow to map failed with  sink row
+          LoggerUtils.logRecord(rows.keySet(), logger);
           mostRecentException = new BigQueryConnectException(failedRowsMap);
           retryCount++;
         } else {
           // throw an exception in case of complete failure
+          LoggerUtils.logRecord(rows.keySet(), logger);
           throw new BigQueryConnectException(failedRowsMap);
```

Comment:
Both of these calls to `LoggerUtils::logRecord` come right before the construction of a `BigQueryConnectException` using the `Map<Long, List<BigQueryError>>` constructor. I wonder if we might add a new constructor to the `BigQueryConnectException` class that allows sink record metadata to be displayed alongside error information for each row, instead of logging record metadata for every failed row here (which may be unnecessary if the only problem is that the connector is writing too many rows to BigQuery per second and needs to be throttled).

Reply:
Changed in favor of trace log all failed records.

Reply:
Done